src/modules/partition/core/PartitionLayout.cpp

```diff
@@ -239,6 +241,33 @@ PartitionLayout::execute( Device* dev,
         {
             currentPartition->fileSystem().setLabel( part.partLabel );
         }
+        if ( !part.partFeatures.isEmpty() )
+        {
+#if defined( WITH_KPMCORE41API )
+            for ( const auto& k : part.partFeatures.keys() )
+            {
+                const auto& v = part.partFeatures.value(k);
+
+                switch (v.type())
+                {
+                case QMetaType::Bool: 
+                    currentPartition->fileSystem().addFeature( { k, v.toBool() } );
+		    break;
+                case QMetaType::LongLong:
+                    currentPartition->fileSystem().addFeature( { k, v.toInt() } );
+		    break;
+                case QMetaType::QString:
+                    currentPartition->fileSystem().addFeature( { k, v.toString() } );
+                default:
+                    cWarning() << "Ignoring feature" << k << "of type" << v.type() << ";"
+                               << "requires type QVariant::bool, QVariant::qlonglong or QVariant::QString.";
+                    break;
+                }
+            }
```

Comment:
Also, I wonder if I can make a change in KPMcore to replace the class [FSFeature](https://github.com/KDE/kpmcore/commit/d24191ebd8ee4a1792003e5bd280cb7679f5e834#diff-4a568e881c09c1f8b464ebde9506d94d) by QVariant; I would simplify the code to the simple call `currentPartition->fileSystem().addFeature( v );`

Reply:
@stikonas opinions about this suggestion for API? That would be a KPMCore 4.2, I guess (or 4.1.1 if you add it, rather than replacing the existing API).

Reply:
If you can keep it ABI compatible, it's fine to go to 4.1.1. See
https://community.kde.org/Policies/Binary_Compatibility_Issues_With_C%2B%2B

Otherwise, it should go to 4.2 (and SOVERSION needs a bump.)

Reply:
If you can keep it ABI compatible, it's fine to go to 4.1.1. See
https://community.kde.org/Policies/Binary_Compatibility_Issues_With_C%2B%2B

Otherwise, it should go to 4.2 (and SOVERSION needs a bump.)